### PR TITLE
cloudtests: provide more flexibility with respect to namespaces

### PR DESCRIPTION
**Pull request by nrainer-materialize**
* Allow specifying namespaces for all resources.
* Make sure that resources access other resources with the correct namespace.
* allow configuring the path to the cockroach setup script

### Motivation
   * This PR refactors existing code.

### CODE REVIEW

File: misc/python/materialize/cloudtest/k8s/cockroach.py [hunk 1]
```diff
@@ -117,8 +122,12 @@ def generate_stateful_set(self) -> V1StatefulSet:
         )
 
 
-COCKROACH_RESOURCES = [
-    CockroachConfigMap(),
-    CockroachService(),
-    CockroachStatefulSet(),
-]
+def create_cockroach_resources(
```
Comment (philip-stoev) [inline misc/python/materialize/cloudtest/k8s/cockroach.py:125]: I think all the `create_` methods are misnamed. They do not create the resources, so the `create_` prefix should be removed altogether.

**Review by philip-stoev — APPROVED** [review]

File: misc/python/materialize/cloudtest/k8s/cockroach.py (hunk 1 above)
Comment (nrainer-materialize) [reply misc/python/materialize/cloudtest/k8s/cockroach.py]: True!

✅ Renamed.

File: misc/python/materialize/cloudtest/k8s/cockroach.py [hunk 2]
```diff
@@ -66,6 +68,9 @@ def __init__(self) -> None:
 
 
 class CockroachStatefulSet(K8sStatefulSet):
+    def __init__(self, namespace: str):
```
Comment (necaris) [inline misc/python/materialize/cloudtest/k8s/cockroach.py:71]: Is this necessary, if the parent class also takes the namespace argument?

**Review by necaris — APPROVED** [review]

File: misc/python/materialize/cloudtest/k8s/cockroach.py (hunk 2 above)
Comment (nrainer-materialize) [reply misc/python/materialize/cloudtest/k8s/cockroach.py:71]: No, you are right. I can remove it in this particular case.

**Pull request merged**